src/metadataDeserialiser.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  AnnotationMetadata,
+  ColonyMetadata,
+  DomainMetadata,
+  MiscMetadata,
+  Metadata,
+  MetadataType,
+} from './types';
+import { METADATA_VERSION } from './constants';
+/*
+ * Stringify functions:
+ * Convert Metadata Object to String (to be stored in IPFS)
+ */
+// get Metadata for Colony type
+export const getStringForMetadataColony = (
+  metaDataArgs: ColonyMetadata,
+): string => {
+  const metadata: Metadata = {
+    version: METADATA_VERSION,
+    name: MetadataType.Colony,
+    data: metaDataArgs,
+  };
+  return JSON.stringify(metadata);
+};
+
+// get Metadata for Domain type
+export const getStringForMetadataDomain = (
+  metaDataArgs: DomainMetadata,
+): string => {
+  const metadata: Metadata = {
+    version: METADATA_VERSION,
+    name: MetadataType.Domain,
+    data: metaDataArgs,
+  };
+  return JSON.stringify(metadata);
+};
+
+// get Metadata for Annotation type
+export const getStringForMetadataAnnotation = (
+  metaDataArgs: AnnotationMetadata,
+): string => {
+  const metadata: Metadata = {
+    version: METADATA_VERSION,
+    name: MetadataType.Annotation,
+    data: metaDataArgs,
+  };
+  return JSON.stringify(metadata);
+};
+
+// get Metadata for Misc type
+export const getStringForMetadataMisc = (
+  miscMetadata: MiscMetadata,
+): string => {
+  const metadata: Metadata = {
+    version: METADATA_VERSION,
+    name: MetadataType.Misc,
+    data: miscMetadata,
+  };
+  return JSON.stringify(metadata);
+};
```

Comment:
I think these are too specific and not scalable. 

If we happen to add a new metadata type, you will need to come back to this lib and do a new release to account for that.

So I would re-write this in a more reusable form:
```ts
export const getStringForMetadata = (
  metadata: Metadata,
  options: Options = {
    name: MetadataType.Misc, // the default type
    version: METADATA_VERSION, // default version, allow for the user to overwrite it if needed
  },
): string => {
  const metadata: Metadata = {
    ...options,
    data: { ...metadata },
  };
  return JSON.stringify(metadata);
};
```

The you can call it something like this:
```ts
getStringForMetadata(
  {
    colonyDisplayName,
    colonyAvatarHash,
    colonyTokens,
    verifiedAddresses,
    isWhitelistActivated,
  },
  {
    name: MetadataType.Colony,
  },
);
```

Reply:
Actually, reading the validation part, setting your own version will not work, so maybe don't include that in the options